engine/servermaster/jobmanager.go

```diff
@@ -84,41 +90,57 @@ type JobManagerImpl struct {
 	// TODO We might add a pending operation queue in the future.
 	jobStatusChangeMu *ctxmu.CtxMutex
 	notifier          *notifier.Notifier[resManager.JobStatusChangeEvent]
+	wg                *errgroup.Group
 }
 
 // CancelJob implements JobManagerServer.CancelJob.
 func (jm *JobManagerImpl) CancelJob(ctx context.Context, req *pb.CancelJobRequest) (*pb.Job, error) {
-	// FIXME: The JobFsm's status may not be consistent with the database.
-	job := jm.JobFsm.QueryOnlineJob(req.Id)
-	if job == nil {
-		// Check if the job is not found.
-		if _, err := jm.frameMetaClient.GetJobByID(ctx, req.Id); pkgOrm.IsNotFoundError(err) {
+	meta, err := jm.frameMetaClient.GetJobByID(ctx, req.Id)
+	if err != nil {
+		if pkgOrm.IsNotFoundError(err) {
 			return nil, ErrJobNotFound.GenWithStack(&JobNotFoundError{JobID: req.Id})
 		}
-		return nil, ErrJobNotRunning.GenWithStack(&JobNotRunningError{JobID: req.Id})
+		return nil, err
```

Comment:
Seems that this error should be wrap as the grpc error by status.Error()

Reply:
If we don't wrap gRPC status, gRPC will automatically wrap it. Maybe we should add unified wrapper as a middleware instead of wrapping it everywhere. This PR won't do it.

```
plain error
apple@dataflow3:~/code/scripts » ./cancel_job.sh c3d79fcf-6e60-4b4d-9261-fcbb806dca62
{"code":2, "message":"test cancel job error"}%

normalized error
apple@dataflow3:~/code/scripts » ./cancel_job.sh c3d79fcf-6e60-4b4d-9261-fcbb806dca62
{"code":2, "message":"JobNotFoundError: {\"JobID\":\"c3d79fcf-6e60-4b4d-9261-fcbb806dca62\"}"}

rpc wrapped error
apple@dataflow3:~/code/scripts » ./cancel_job.sh c3d79fcf-6e60-4b4d-9261-fcbb806dca62
{"code":10, "message":"JobNotFoundError: {\"JobID\":\"c3d79fcf-6e60-4b4d-9261-fcbb806dca62\"}"}
```